### DISCUSSION TITLE
Changes to be made in overlay to modify the BNN

### DISCUSSION THREAD

**Discussion by shobhitt10** [Xilinx/finn#470] [Q&A]
Hi, I want to implement some research paper on PYNQ i.e. need to modify the bnn/qnn architecture (say the activation function, layers etc.). So, what changes should be made in the vivado block design to generate bitstream? and what all things will remain same (e.g. AXI)

Also, how can I get the block diagram of bnn in vivado project to get bitstreams? (tried with .tcl file from http://192.168.2.99:9090/tree/BNN-PYNQ-master/bnn/bitstreams/pynqZ1-Z2/ but it had version issues)

**Comment by lloo099** [comment]
U may follow BNN-PYNQ github branch
